engine/org.eclipse.birt.report.engine.emitter.wpml/src/org/eclipse/birt/report/engine/emitter/wpml/EmitterServices.java

```diff
@@ -0,0 +1,418 @@
+		}
+
+		if (birtContent != null && birtContent.getReportContent() != null && value == null) {
+			value = getReportDesignConfiguration(birtContent.getReportContent(), name);
+		}
+
+		if (value instanceof Number) {
+			result = ((Number) value).intValue();
+		} else if (value != null) {
+			try {
+				result = Integer.parseInt(value.toString());
+			} catch (Exception ex) {
+			}
+		}
+
+		return result;
+	}
+
+	/**
+	 * Convert an Object to a boolean, with quite a few options about the class of
+	 * the Object.
+	 *
+	 * @param value        A value that can be of any type.
+	 * @param defaultValue Value to return if value is null.
+	 * @return true if value in some way represents a boolean TRUE value.
+	 */
+	public static boolean booleanOption(Object value, boolean defaultValue) {
+		if (value != null) {
+			if (value instanceof Boolean) {
+				return ((Boolean) value).booleanValue();
+			}
+			if (value instanceof Number) {
+				return ((Number) value).doubleValue() != 0.0;
+			}
+			if (value != null) {
+				return Boolean.parseBoolean(value.toString());
+			}
+		}
+		return defaultValue;
+	}
+
+	/**
+	 * Get the boolean value of a user property
+	 *
+	 * @param propertyName user property name
+	 * @param defaultValue default of the user property
+	 * @return the user property result of boolean
+	 */
+	public static boolean booleanOption(String propertyName, boolean defaultValue) {
+		return booleanOption(null, propertyName, defaultValue);
+	}
+
+	/**
+	 * Get the boolean value of a user property
+	 *
+	 * @param birtContent  element of the report
+	 * @param propertyName user property name
+	 * @param defaultValue default of the user property
+	 * @return the user property result of boolean
+	 */
+	public static boolean booleanOption(IContent birtContent, String propertyName, boolean defaultValue) {
+		Object value = getUserProperty(birtContent, propertyName);
+		return booleanOption(value, defaultValue);
+	}
+
+	/**
+	 * Get the string value of a user property
+	 *
+	 * @param birtContent  element of the report
+	 * @param propertyName user property name
+	 * @param defaultValue default of the user property
+	 * @return the user property result as string
+	 */
+	public static String stringOption(IContent birtContent, String propertyName, boolean defaultValue) {
+		String stringValue = null;
+		Object value = getUserProperty(birtContent, propertyName);
+		if (value != null) {
+			stringValue = value.toString();
+		}
+		return stringValue;
+	}
+
+	/**
+	 * Get the object of the user property
+	 *
+	 * @param birtContent report element, e.g. a IListContent
+	 * @param propName    property name
+	 * @return user property object independent of the type
+	 */
+	protected static Object getUserProperty(IContent birtContent, String propName) {
+		Object value = null;
+
+		Map<String, Object> userprops = birtContent.getUserProperties();
+		if (userprops != null) {
+			value = userprops.get(propName);
+		}
+		if (value == null) {
```

Comment:
This change cannot be done because this is a cascading of different checks if the value is "null",
so we have the following checks:
1. birtContent.getUserProperties()
2. designElem.getUserProperties()
3. birtContent.getReportContent().getDesign().getUserProperties()